### PR TITLE
[WIP] Migrate data-explorer to typescript

### PR DESCRIPTION
**Pull request by BenRussert**
I'll close the other PR as this one has our collaborative work so far. We'll need to rebase once its at a good point for merging to master.

### CODE REVIEW

File: packages/transform-dataresource/package.json [hunk 1]
```diff
@@ -50,5 +46,14 @@
     "semiotic": "^1.16.0",
     "styled-jsx": "^3.1.0",
     "tv4": "^1.3.0"
+  },
+  "devDependencies": {
```
Comment (jdetle) [inline packages/transform-dataresource/package.json:50]: I believe these are meant to exist at the root level? Still not sure about this.

**Comment by emeeks** [comment]
When I take a look at this in VS Code everything seems fine but when I run `yarn build` I see a host of TypeScript errors. Any idea why this is the case? Is there something I should be doing with my VS Code config?

**Comment by captainsafia** [comment]
Based on the discussion from Slack, I'll be rebasing this PR on the command line and merging it into the nteractors/jan/dx-ts-migration branch. Then we can all submit small PRs related to the data explorer migration to that branch.

> rgbkrk [10:13 AM]
@captainsafia is going to make a branch for the dx -> ts conversion PR
and people can submit PRs to that branch

> captainsafia [10:14 AM]
> Yep.
I'm gonna go ahead and resolve the conflicts on the open PR from the command line.
Then I will merge that PR into the new branch I made.
Then everyone can submit smaller PRs with TS fixes to that branch.
The branch is https://github.com/nteract/nteract/tree/nteractors/jan/dx-ts-migration.

**Comment by rgbkrk** [comment]
Ok cool, I'll start making my styled-components PRs to the new branch.

**Comment by captainsafia** [comment]
Closing in favor of https://github.com/nteract/nteract/pull/3940 which includes all commits from this one and a rebase for conflicts.

**Pull request closed without merging**